dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_fields.tsx

```diff
@@ -40,10 +46,14 @@ export const DataConfigPanelFields = ({
   handleServiceEdit,
 }: DataConfigPanelFieldProps) => {
   const isAggregation = sectionName === AGGREGATIONS;
-  const isHeatMapAddButton = (name: string) => {
-    if (!list || !isArray(list) || visType !== VIS_CHART_TYPES.HeatMap) return false;
-    return name === AGGREGATIONS ? list.length >= 1 : list.length >= 2;
+  const hideAddButton = (name: string) => {
+    if (!isArray(list) || !HIDE_ADD_BUTTON_VIZ_TYPES.includes(visType)) return false;
```

Comment:
As above.  I would spend over 1/2 hour reading this code and still have the logic description wrong.

Reply:
This function would be modified further in the coming weeks with the new implementation of breakdown for line and scatter. For now have mentioned comments to justify the logic.